include/deal.II/numerics/vector_tools_integrate_difference.templates.h

```diff
@@ -882,6 +882,8 @@ namespace VectorTools
                                                 q,
                                                 update_flags);
 
+      difference = 0;
+
```

Comment:
changes like to this file, or converting for loops to range-based are independent of this patch -- you'll make everyone's lives easier if you extract these to separate patches because then the one you're really after becomes smaller and easier to review.